bindings/python/src/session.cpp

```diff
@@ -388,6 +388,20 @@ namespace
     }
 #endif // TORRENT_ABI_VERSION
 
+    void alert_notify(object cb)
+    {
+        if (cb)
```

Comment:
is it OK to check to see if `cb` is set without holding the GIL?

Reply:
I'm not sure. Let me see what I can find out. I have read in the past that whenever you deal with Python objects you should always acquire the GIL, so maybe it would be prudent to just do that in this case. I do worry a bit about blocking the libtorrent main thread while waiting to acquire the lock though.

Reply:
Oh whoops. We grab the gil anyways if cb is not None. I'll just move the lock to the top of the function just to be safe.